coderd/httpapi/queryparams.go

```diff
@@ -233,7 +246,7 @@ func ParseCustomList[T any](parser *QueryParamParser, vals url.Values, def []T,
 func parseQueryParam[T any](parser *QueryParamParser, vals url.Values, parse func(v string) (T, error), def T, queryParam string) (T, error) {
 	parser.addParsed(queryParam)
 	// If the query param is required and not present, return an error.
-	if parser.RequiredParams[queryParam] && (!vals.Has(queryParam)) {
+	if parser.RequiredParams[queryParam] && (!vals.Has(queryParam) || vals.Get(queryParam) == "") {
```

Comment:
Yeah I had mixed feelings about this; it felt surprising to me that I could write `Required("client_id")` and it could come through empty.  My thinking was that if I want to require a query param I also want a value.

I did check existing usages; it is used for `reconnect` on the terminal (which is invalid if blank) and for `start_time` and `end_time` on an insights endpoint which also cannot be blank (it will fail to be parsed into a time).

But at the same time I see the distinction between required and empty.  An empty query value is probably only useful as a boolean where the presence of the key means `true`, and I could see requiring a `true` for something like `?acceptTerms` or something like that before you can use the endpoint.

So let me rename this to `RequiredNotEmpty`, that will leave room for `Required` if we do need that some day.

Reply:
Some new ones popped up after I merged main: `template_id`, `workspace_id`, and `agent_id` on some endpoints, they all make database queries with these values so I think they would just return "no rows" or a 404 or whatever, so this change will make it return "invalid query params, template_id is required" instead which I think is fair, but it does change the semantics a little from a 404 to a 400.